### DISCUSSION TITLE
Question on Input Formats

### DISCUSSION THREAD

**Discussion by rbelshe** [FLAMEGPU/FLAMEGPU2#770] [Q&A] [answered]
Hello, 

My team and I recently downloaded and installed Flame GPU 2, and we are curious about running the examples with XML or JSON input files.  For example, we tried using the 0.xml that worked with the game of life examples with Flame GPU 1.5.  Invoking the code with:
/packages/7x/flamegpu/2.0/bin/game_of_life --in 0.xml -s 5 -d 0 -r 1

Which then generates this error:

terminate called after throwing an instance of 'std::out_of_range'

  what():  _Map_base::at

Aborted

Have the input formats changed? If anyone can insight on this that would be most appreciated. 

Thanks!

**Comment by Robadob** [comment, marked as answer]
Whilst FLAMEGPU2 xml input files are backwards compatible, some of the models have changed slightly. In this case I believe the name and format of the agent variables have changed in the FLAMEGPU Game of Life model.

You should be able to run the model without an input file, as the FLAMEGPU2 model has built in initialisation to generate an initial state. If you require an input file, you can generate a sample one by adding a call to `CUDASimulation::exportData()` ([example here](https://github.com/FLAMEGPU/FLAMEGPU2/blob/189d4496493cd949e97b6e7346cdb0ab3f1181d6/examples/circles_spatial3D/src/main.cu#L189)) which will generate an output file that should be compatible as an input file.

> Which then generates this error: ...

That error isn't very useful, so we've patched it [here](https://github.com/FLAMEGPU/FLAMEGPU2/pull/765), the fix should be merged shortly. It was caused by the input file containing an agent state not found in the new model.

> Have the input formats changed? If anyone can insight on this that would be most appreciated.

Our documentation of input files is sorely lacking, I've put together a draft [here](https://github.com/FLAMEGPU/FLAMEGPU2-docs/pull/71) which will hopefully be merged soon.

Please let me know if you have any further issues.
Thanks

**Reply by Robadob** [reply]
Both of the pull requests linked in the above message have now been merged. So [docs.flamegpu.com](https://docs.flamegpu.com/) should have the updated IO docs in 5-10 minutes when CI has processed it. Similarly, if you build a new copy of FLAMEGPU2 from `master`, it will provide a more specific exception if you attempt to load that particular input file.
